### PR TITLE
Switch to docker CI

### PR DESCRIPTION
**Pull request by Timple**
Should work when https://github.com/tue-robotics/tue-env/pull/98 is fixed.

Last two commits here are tests:
https://travis-ci.org/tue-robotics/tue_robocup/builds

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,40 +1,19 @@
...
-  # install travis' package dependencies
-  - cd "$TUE_SYSTEM_DIR/src"
-  - rm *
-  - tue-get update ros-$REPOSITORY_NAME
-  - ls -l
-
-  # go to the catkin workspace
-  - cd "$TUE_SYSTEM_DIR"
+before_install:
+  - wget https://raw.githubusercontent.com/tue-robotics/tue-env/master/ci/install-package.sh
```
Comment (LoyVanBeek) [inline .travis.yml]: Can we make this build trigger as well when tue-env is updated? 
Via http://eng.rightscale.com/2015/04/27/dependent-builds-in-travis.html or https://github.com/mernst/plume-lib/blob/master/bin/trigger-travis.sh

Comment (Timple) [inline .travis.yml]: We can! Feel free to put in a new issue.

Especially the upstream is quite hard to parse in our current setup. Downstream is doable (and actually already performed by tue-get).

Comment (LoyVanBeek) [inline .travis.yml]: Upstream could be done with ```rospack depends-on``` for ROS packages.

File: .travis.yml [hunk 2]
```diff
@@ -1,40 +1,19 @@
...
+  - wget https://raw.githubusercontent.com/tue-robotics/tue-env/master/ci/build-package.sh
+  # Determine packages modified in this commit
+  - PACKAGES=`git diff-tree --name-only $TRAVIS_COMMIT | xargs ls -dl 2>/dev/null |  grep "^d" | awk '{print $NF}' `
 
-script:
-  - catkin build
-  - catkin run_tests # and run them
```
Comment (LoyVanBeek) [inline .travis.yml]: Run ```catkin_run_tests``` in the docker image

File: .travis.yml (hunk 1 above)
Comment (LoyVanBeek) [inline .travis.yml]: Issue filed: #314.

File: .travis.yml (hunk 2 above)
Comment (LoyVanBeek) [inline .travis.yml]: Something like 
```docker run -it -d tuerobotics/tue-env catkin run_tests```

Comment (Timple) [inline .travis.yml]: That is a good point!

https://github.com/tue-robotics/tue-env/pull/99
